### DISCUSSION TITLE
The getFullList view columns are getting re-sorted alphabetically

### DISCUSSION THREAD

**Discussion by dougwatson** [pocketbase/pocketbase#2923] [Q&A]
Hi, I am trying to write reports out dynamically into a javascript spreadsheet but the problem I'm facing is that the columns are getting reordered (alphabetically) when I query them out with pb.collection(viewTable).getFullList(options). I would like to get the real column order (as in the original view query) so I don't have to hardcode the column order in the client.

Is there any workaround for this?  I have written several view tables in pocketbase that I want to display in the web browser but it is loosing the column order and sending the columns back in alphabetic order. I have tested with curl to verify that this is not just a browser display problem.

curl 'http://localhost:8080/api/collections/my_report_view/records?page=1&perPage=200'

The months were in correct order in the view sql query, but they get reordered in the output: 

{"page":1,"perPage":200,"totalItems":10,"totalPages":1,"items":[{"Apr":0,"Aug":0,"Dec":0,"Feb":0,"Jan":0,"Jul":0,"Jun":0,"Mar":0,"May":0,"Nov":0,"Oct":0,"Sep":0,

Thanks,
Doug

**Comment by ganigeorgiev** [comment]
No, at the moment there is no easy way to do it without implementing our own json encoder/decoder.
`encoding/json`, and Go maps in general, don't preserve the insertion order and you shouldn't rely on it (see for example https://github.com/golang/go/issues/27179).

As a workaround, if you want sorted results you can define a helper `order`/`rank`/etc. field, or maybe better - have the months as padded 2 digits number, and sort based on the field value (eg. `&sort=-order`).

**Reply by dougwatson** [reply]
Thanks for the quick reply! that helped me understand the problem better and I think I came up with a decent work-around.

In case someone comes across this same issue, here is how I figured out how to restore the column order of a json result set.

```
var jsonData = await pb.collection(tableName).getFullList(options); //fetch the data (it's a view)

//set collection back to tableName, it may have been changed in the getFullList call
$activeCollection = $collections.find((collection) => collection.name === tableName); 

let orderedJsonData = [];
for(let i = 0; i < $activeCollection.schema.length; i++) {
  // create new objects with properties in the order of $activeCollection.schema
  orderedJsonData = jsonData.map(item => {
    let orderedItem = {};
    for (let j = 0; j < $activeCollection.schema.length; j++) {
      let schemaName = $activeCollection.schema[j].name;
      orderedItem[schemaName] = item[schemaName];
    }
    return orderedItem;
  });
}
```

Then you will be able to use the **orderedJsonData** just like the original **jsonData** but the columns will be in the right order.

One oddity is that in my code, for some reason I needed to reset the activeCollection to the tableName (actually it's a view) after running getFullList query, if not it would get set to the wrong collection. But after I do that it works fine.
